src/ophyd_async/epics/adandor/_andor_controller.py

```diff
@@ -0,0 +1,65 @@
+import asyncio
+
+from ophyd_async.core import (
+    AsyncStatus,
+    DetectorController,
+    DetectorTrigger,
+    TriggerInfo,
+    set_and_wait_for_value,
+)
+from ophyd_async.epics.adcore import (
+    set_exposure_time_and_acquire_period_if_supplied,
+    stop_busy_record,
+)
+
+from ._andor_io import Andor2DriverIO, Andor2TriggerMode, ImageMode
+
+_MIN_DEAD_TIME = 0.1
+_MAX_NUM_IMAGE = 999_999
+
+
+class Andor2Controller(DetectorController):
+    def __init__(
+        self,
+        driver: Andor2DriverIO,
+    ) -> None:
+        self._drv = driver
+        self._arm_status: AsyncStatus | None = None
+
+    def get_deadtime(self, exposure: float | None) -> float:
+        return _MIN_DEAD_TIME + (exposure or 0)
+
+    async def prepare(self, trigger_info: TriggerInfo):
```

Comment:
Could we not subclass `TriggerInfo` like eiger does, making an `Andor2TriggerInfo` which contains `trigger: Andor2TriggerMode` instead of `trigger: DetectorTrigger`?